core/src/main/java/org/apache/calcite/plan/volcano/IterativeRuleQueue.java

```diff
@@ -101,51 +63,31 @@
    */
   @Override public boolean clear() {
     boolean empty = true;
-    for (PhaseMatchList matchList : matchListMap.values()) {
-      if (!matchList.queue.isEmpty() || !matchList.preQueue.isEmpty()) {
-        empty = false;
-      }
-      matchList.clear();
+    if (!matchList.queue.isEmpty() || !matchList.preQueue.isEmpty()) {
+      empty = false;
     }
+    matchList.clear();
     return !empty;
   }
 
-  /**
-   * Removes the {@link PhaseMatchList rule-match list} for the given planner
-   * phase.
-   */
-  public void phaseCompleted(VolcanoPlannerPhase phase) {
-    matchListMap.get(phase).clear();
-  }
-
   /**
    * Adds a rule match. The rule-matches are automatically added to all
-   * existing {@link PhaseMatchList per-phase rule-match lists} which allow
+   * existing {@link MatchList per-phase rule-match lists} which allow
    * the rule referenced by the match.
    */
   public void addMatch(VolcanoRuleMatch match) {
     final String matchName = match.toString();
-    for (PhaseMatchList matchList : matchListMap.values()) {
-      Set<String> phaseRuleSet = phaseRuleMapping.get(matchList.phase);
-      if (phaseRuleSet != ALL_RULES) {
-        String ruleDescription = match.getRule().toString();
-        if (!phaseRuleSet.contains(ruleDescription)) {
-          continue;
-        }
-      }
 
-      if (!matchList.names.add(matchName)) {
-        // Identical match has already been added.
-        continue;
-      }
+    if (!matchList.names.add(matchName)) {
+      // Identical match has already been added.
+      return;
+    }
 
-      LOGGER.trace("{} Rule-match queued: {}", matchList.phase.toString(), matchName);
+    LOGGER.trace("{} Rule-match queued: {}", matchList.toString(), matchName);
```

Comment:
You don't want to log matchlist. Remove the first argument.

Reply:
> You don't want to log matchlist. Remove the first argument.

Thanks for your suggestion, Revised.

Reply:
> You don't want to log matchlist. Remove the first argument.

Thanks for your suggestion, Revised.